File: packages/autonavi_maps_flutter/ios/Classes/AutonaviMapsFlutterPlugin.swift
import Flutter
import AMapFoundationKit

public class AutonaviMapsFlutterPlugin: NSObject, FlutterPlugin {

    public static func register(with registrar: FlutterPluginRegistrar) {
        // Privacy compliance — must be called before any AMap SDK initialisation.
        AMapServices.sharedServices()?.updatePrivacyShow(.didShow, privacyInfo: .didContain)
        AMapServices.sharedServices()?.updatePrivacyAgree(.didAgree)

        let factory = AMapFactory(messenger: registrar.messenger())
        registrar.register(factory, withId: "plugins.autonavi.flutter/amap_map")
    }
}

File: packages/autonavi_maps_flutter/android/src/main/kotlin/com/autonavi/maps/flutter/AutonaviMapsFlutterPlugin.kt
package com.autonavi.maps.flutter

import android.content.Context
import com.amap.api.maps.MapsInitializer
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.embedding.engine.plugins.activity.ActivityAware
import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding
import io.flutter.plugin.common.BinaryMessenger

class AutonaviMapsFlutterPlugin : FlutterPlugin, ActivityAware {

    private var flutterPluginBinding: FlutterPlugin.FlutterPluginBinding? = null

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        flutterPluginBinding = binding
        val context = binding.applicationContext

        // Privacy compliance — must be called before any AMap SDK initialization
        MapsInitializer.updatePrivacyShow(context, true, true)
        MapsInitializer.updatePrivacyAgree(context, true)

        binding.platformViewRegistry.registerViewFactory(
            "plugins.autonavi.flutter/amap_map",
            AMapFactory(binding.binaryMessenger, context),
        )
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        flutterPluginBinding = null
    }

    override fun onAttachedToActivity(binding: ActivityPluginBinding) {}
    override fun onDetachedFromActivityForConfigChanges() {}
    override fun onReattachedToActivityForConfigChanges(binding: ActivityPluginBinding) {}
    override fun onDetachedFromActivity() {}
}
